src/main/java/com/Others/FastPower.java

```diff
@@ -0,0 +1,37 @@
+package src.main.java.com.Others;
+
+import java.math.BigInteger;
+
+/**
+ * We may calculate power with loops, but what if the index is too large ?
+ * FastPower aims to calculate quickly in this circumstances with time complexity O(log k),
+ * where k is the index.
+ *
+ * @author DDullahan
+ */
+public class FastPower {
+    public static BigInteger calculate(BigInteger n, BigInteger k, BigInteger mod) {
+        BigInteger ans = BigInteger.ONE;
+        while (!k.equals(BigInteger.ZERO)) {
+            int odd = k.mod(new BigInteger("2")).compareTo(BigInteger.ZERO);
+            if(odd == 1){
```

Comment:
The result of [`compareTo`](https://docs.oracle.com/javase/7/docs/api/java/math/BigInteger.html#modPow(java.math.BigInteger,%20java.math.BigInteger)) should only be compared with 0, there is no guarantee that it will be 1, only that it may be negative, 0 or positive.